test/integration/integration_test.exs

```diff
@@ -49,13 +49,21 @@ defmodule KafkaEx.Integration.Test do
     assert consumer_group_update_interval == 30000
   end
 
-  test "create_worker provides a default consumer_group of 'kafka_ex'" do
```

Comment:
I think the example you showed is why disabling `consumer_group` is based off of `consumer_group` configuration also requiring this to be explicitly configured means we require users to configure this to get the magical offset management. This might be better as explicit is better than implicit, but then we have to document this new behavior we are introducing, also we should change https://github.com/dantswain/kafka_ex/blob/fix_parameter_merging/lib/kafka_ex.ex#L38-L39 to use `Application.fetch_env!` instead of `Application.get_env` since this is a require param